tools/bench/perf_probe.py

```
@@ -0,0 +1,143 @@
+
+
+def recursion_depth_reached(code: str, target: int) -> tuple[bool, int, float]:
+    env = _make_env()
+    old_stdout = sys.stdout
+    sys.stdout = io.StringIO()
+    try:
+        t = time.perf_counter()
+        try:
+            run_source(code, env)
+            return True, target, time.perf_counter() - t
+        except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)